js/packages/mobile-wallet-adapter-protocol/src/createMobileWalletProxy.ts

```diff
@@ -31,8 +31,8 @@ export default function createMobileWalletProxy<
                     const { method, params } = handleMobileWalletRequest(p, inputParams, protocolVersion);
                     const result = await protocolRequestHandler(method, params) as Awaited<ReturnType<MobileWallet[TMethodName]>>;
                     // if the request tried to sign in but the wallet did not return a sign in result, fallback on message signing
-                    if (method === 'authorize' && params?.hasOwnProperty('sign_in_payload') && !result.hasOwnProperty('sign_in_result')) {
-                        (result as any)['sign_in_result'] = signInFallback(
+                    if (method === 'authorize' && (params as any).sign_in_payload && !(result as any).sign_in_result) {
```

Comment:
why not `params?.sign_in_payload` and `result?.sign_in_result`?

Reply:
because the type of param is generic (boxed to ReurnType<MobileWallet>), so you cannot do this.